torch/_inductor/lowering.py

```diff
@@ -4198,7 +4299,6 @@ def loader(index, reduction_index):
         assert len(reduction_index) == len(reduced_idx)
         if keepdims:
             assert len(index) == len(size)
-            assert all(index[i] == 0 for i in reduced_idx)
```

Comment:
I think an user recently reported this as a bug... is this something we're fixing?

Reply:
Yea, I confirmed with @peterbell10, this check doesn't make sense with masked loads